docs/application-connector-compass-mode/02-04-application-broker.md

```diff
@@ -8,45 +8,60 @@ The Application Broker (AB) workflow consists of the following steps:
+
+![API ServicePlan](./assets/005-AB-API-service-plan.svg)
+
+### Generating API credentials
+
+This section describes in detail how the API credentials are generated.
+
+1. User creates a ServiceInstance.
+2. Service Catalog sends a provisioning request to the Application Broker.
+3. Application Broker requests credentials for a ServiceInstance. The request is processed through the Director proxy exposed by Runtime Agent to Director.
+>**NOTE:** Director proxy is secured by [Istio RBAC](https://github.com/kyma-project/kyma/blob/master/resources/compass-runtime-agent/templates/istio-rbac.yaml) and only Application Broker can access it.
+
+4. Application Broker polls credential status with 20 minutes timeout. The polling is completed when the credential status is either `SUCCEEDED` or `FAILED`.
+
+![Credentials flow](./assets/api-credentials-flow.svg)
 
-![API Service Class](./assets/005-AB-API-service-class.svg)
+Application Broker uses these credentials during the binding action. To use another set of credentials, you must create a new ServiceInstance.
 
-## Provisioning and binding for an event ServiceClass
+## Provisioning and binding for an event ServicePlan
 
-This ServiceClass has a **bindable** parameter set to `false` which means that after provisioning a ServiceClass in the Namespace, given events are ready to use for all services. The provisioning workflow for an event ServiceClass consists of the following steps:
+This ServicePlan has a **bindable** parameter set to `false` which means that after provisioning a ServiceInstance in a Namespace, given events are ready to use for all services. The provisioning workflow for an event ServicePlan consists of the following steps:
 
-1. Select a given event ServiceClass from the Service Catalog.
-2. Provision this ServiceClass by creating a ServiceInstance in the given Namespace.
+1. Select a given ServiceClass with an event ServicePlan from the Service Catalog.
+2. Provision this ServiceClass with the chosen event ServicePlan by creating a ServiceInstance in the given Namespace.
 3. During the provisioning process, the [EventActivation](/components/application-connector/#custom-resource-event-activation) resource is created together with the ServiceInstance. This resource allows you to create a Trigger from the Kyma Console.
 4. The Application sends an event to the Application Connector.
-5. When the Trigger receives the event, it triggers the Function based on the paramertes defined for a Trigger.
+5. When the Trigger receives the event, it triggers the Function based on the parameters defined for a Trigger.
 
 For more information, read about [event processing and delivery](/components/event-mesh/#details-event-processing-and-delivery).
 
-![Event Service Class](./assets/006-AB-event-service-class.svg)
+![Event ServicePlan](./assets/006-AB-event-service-plan.svg)
```

Comment:
Shouldn't we add arrows in the diagram to steps 3, 4, and 5?

Reply:
I'd add an arrow to step 5, as Trigger is the subject here. In 3 and 4, all the magic happens in the background - that's why we've decided to use the dashed line.

Reply:
Fine by me :)